### PR TITLE
Resource updates/refactor

### PR DESCRIPTION
**Pull request by MRossol**
- change dsets property to datasets for clarity
- add __iter__
- add __contains__

### CODE REVIEW

File: reV/handlers/resource.py [hunk 1]
```diff
@@ -259,6 +260,27 @@ def __getitem__(self, keys):
...
+        dset = self.datasets[self._i]
+        self._i += 1
+
+        return dset
+
+    def __contains__(self, dset):
+        test = dset in self.datasets
+        if not test:
+            msg = "{} does not exist in {}".format(dset, self)
+            raise HandlerKeyError(msg)
```
Comment (grantbuster) [inline reV/handlers/resource.py]: I think this is supposed to return False instead of raise an exception?

File: reV/handlers/resource.py [hunk 2]
```diff
@@ -305,7 +327,7 @@ def h5(self):
         return h5
 
     @property
-    def dsets(self):
```
Comment (grantbuster) [inline reV/handlers/resource.py]: Thoughts on maintaining both  for backwards compatibility? I dont think keeping two aliases around would be a bad idea, especially when its a minor issue like abbreviation.

File: reV/handlers/resource.py (hunk 1 above)
Comment (MRossol) [reply reV/handlers/resource.py]: hmm, all I changed was the property name...

File: reV/handlers/resource.py (hunk 2 above)
Comment (MRossol) [reply reV/handlers/resource.py]: good call. I'll add dsets back in

**Pull request merged**